query/api/src/main/java/org/codice/ddf/admin/api/poller/EnumValuePoller.java

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ **/
+package org.codice.ddf.admin.api.poller;
+
+import java.util.List;
+
+import org.codice.ddf.admin.api.fields.EnumValue;
+
+public interface EnumValuePoller<T> {
```

Comment:
I agree it is a pretty rare use case to want an `EnumValue` that is not of type String type AND needs to be dynamic. I actually wrote it with the generic specified as a String but changed it back to be consistent with the `EnumField` and `EnumValue` interfaces. We should decide whether `EnumValue` should be generic and update this accordingly.